### ISSUE TITLE
Search should find also terms with diacritics, e.g. 'deja vu' -> 'déjà vu'

### ISSUE THREAD

**Issue by osma** [NatLibFi/Skosmos#313]
## At which URL did you encounter the problem?
http://finto.fi/yso/fi/search?clang=en&q=deja+vu

## What steps will reproduce the problem?
1. Search YSO in English for "deja vu" (without diacritics)

## What is the expected output? What do you see instead?

Expected to find the concept "déjà vu" but it cannot be found because of the diacritics. The terms are matched exactly in the current implementation.

This could be implemented after #273 by making use of a better text index configuration that takes into account diacritics (i.e., stores a stripped variant of the term in the index and uses that for matching). However, this likely requires implementing yet another Analyzer for jena-text.

Related to https://github.com/NatLibFi/Finto-data/issues/11 though that attempts to solve the problem by adding hiddenLabel properties into the data instead.

**Comment by tfrancart** [comment]
:+1: 
Would love to see this implemented. An extra index analyzer, similar to text:LowerCaseKeywordAnalyzer, would be the way to go. SolR provides such an Filter : solr.ASCIIFoldingFilterFactory

**Comment by osma** [comment]
Yes, I agree. But as I said above this requires implementing #273 first, because the current SPARQL queries rely on both jena-text and string matches on literals in SPARQL. #273 is scheduled for the next development cycle (1.4), so this could be done in the 1.5 cycle at the earliest.

Unfortunately the Solr support in jena-text has fallen behind the Lucene support, so we probably cannot use the Solr filter directly. Instead a similar Lucene filter needs to be added to jena-text.

**Comment by tfrancart** [comment]
An idea of when 1.5 is schedule ?
This is a real showstopper for thesaurus in french (and spanish) that use diacritics a lot - while users tend to forget them when searching.
What is the workaround you currently recommend ? pre-compute labels without diacritics and store them as hidden labels ?

**Comment by osma** [comment]
We don't have a fixed release schedule. Our ideal is 1 month release cycles but in practice it has been more like 2-3 months. Depends a lot on what else is going on as well.

I understand the problem, it's just not that easy to fix. Adding hiddenLabels would work.

**Comment by osma** [comment]
I made an analyzer for jena-text which should be suitable for this and submitted it for inclusion in Jena:
https://github.com/apache/jena/pull/96
https://issues.apache.org/jira/browse/JENA-1058

**Comment by osma** [comment]
Now that I think of it, after #273 the text search would rely exclusively on the jena-text index. Then it would be a matter of configuring the text index for accent folding. There may not actually be anything to implement in Skosmos code in 1.5, this could already be done with 1.4 as long as a suitable version of jena-text (e.g. my patched version above) is used and properly configured.

**Comment by osma** [comment]
This could now be done on the jena-text side with ConfigurableAnalyzer, which was just merged to Jena trunk and should be released with Jena 3.0.1 when it comes out.
See https://issues.apache.org/jira/browse/JENA-1062 and https://github.com/apache/jena/pull/97

**Comment by osma** [comment]
Now that #273 has been implemented and searches are only based on jena-text, this should be a matter of configuring jena-text to use a [ConfigurableAnalyzer](https://jena.apache.org/documentation/query/text-query.html#configurableanalyzer) which includes ASCIIFoldingFilter in the pipeline.

Unfortunately jena-text in Jena 3.0.1 is buggy ([JENA-1093](https://issues.apache.org/jira/browse/JENA-1093)) so to really test this properly would require using a very recent Jena/Fuseki snapshot (I fixed JENA-1093 today in Jena master).

Anyone want to test? @tfrancart @danmichaelo ? Just put something like this in the jena-text configuration for all the SKOS label properties (prefLabel, altLabel, hiddenLabel), recreate the text index, and try searching in Skosmos for terms with accents or other diacritics:

```
text:analyzer [
  a text:ConfigurableAnalyzer ;
  text:tokenizer text:KeywordTokenizer ;
  text:filters (text:ASCIIFoldingFilter, text:LowerCaseFilter)
]
```

**Comment by osma** [comment]
I've tested that with a suitable combination of a recent Fuseki snapshot and a ConfigurableAnalyzer and ASCIIFoldingFilter, this is now possible. I wrote up some documentation about [alternative text analyzers](https://github.com/NatLibFi/Skosmos/wiki/TextAnalysisConfiguration) in the wiki.

Since at least the basic stuff is done and there is nothing more obvious to do on the Skosmos side, I will close this issue and mark it as done for 1.4.
